### ISSUE TITLE
Missing mirror URL on project dashboard

### ISSUE THREAD

**Issue by somera** [go-gitea/gitea#28203]
### Description

Since I upgraded to 1.21.0 I see more and more mirrors with no **Mirror URL**

![image](https://github.com/go-gitea/gitea/assets/8334250/852c009f-a93d-41cb-9f5e-8179df7c1a23)

The sync for this mirror is working.

In PostgreSQL I see the URL

![image](https://github.com/go-gitea/gitea/assets/8334250/744996f7-0899-4c9c-b7c8-bed5155269a3)

I checked ~200 projects. In 24 the mirror url is not visible. some days ago I deleted one mirror and added it again and I saw the mirror url. But today I can't see it in project dashboard.

### Gitea Version

1.21.0

### Can you reproduce the bug on the Gitea demo site?

No

### Git Version

2.43.0

### Operating System

Ubuntu 20.04.x

### How are you running Gitea?

gitea-1.21.0-linux-amd64

### Database

PostgreSQL

**Comment by lunny** [comment]
Could you find any error logs?

**Comment by somera** [comment]
@lunny. No.

**Comment by lunny** [comment]
I think it maybe a translation problem. Could you try English?

**Comment by somera** [comment]
As I wrote. No -> I can't see any errors in the gitea logs.

The screenshot which I posted was just from the browser. This was probably confusing.

**Comment by lunny** [comment]
Yes. To confirm my guess. Could you try to switch English from the footer right corner and take a look again?

**Comment by somera** [comment]
I try this later. Im upgrading now to PostgreSQL 16 and Ubuntu 22.04.

**Comment by somera** [comment]
I did it. Same problem
![image](https://github.com/go-gitea/gitea/assets/8334250/08dfb461-9e44-4f3b-b313-0c380bcee65e)

**Comment by somera** [comment]
I found at the moment 44 projects with this issue. But I have no time to check all mirrors.

**Comment by somera** [comment]
~80 mirrors now affected.

**Comment by somera** [comment]
@lunny no changes here with 1.21.1

**Comment by somera** [comment]
@lunny any updates here? The problem makes it a little difficult for me to work with Gitea.

**Comment by somera** [comment]
@lunny I found the problem. ~17% of mirror are affected on my instance. And the problem ist, that in mirrors table the remote_address column is empty. I will fix it by setting it. But there is an bug, cause I migrated new mirrors and some days later the url was gone.

The question is: why is this empty?

**Comment by somera** [comment]
After my fix in the DB three days ago nothing changes. It's looks good.

@lunny if you think there is nothig to fix, than this issue can be closed.

**Comment by yp05327** [comment]
Maybe related to the migration in #26952

**Comment by lunny** [comment]
> After my fix in the DB three days ago nothing changes. It's looks good.
> 
> @lunny if you think there is nothig to fix, than this issue can be closed.

Let's leave the issue open since it's possible a bug. And how did you do the workaround?

**Comment by yp05327** [comment]
It seems that this issue is caused by the migration, as `remote_address` column is empty but `original_url` is not.

**Comment by somera** [comment]
> Let's leave the issue open since it's possible a bug. And how did you do the workaround?

I set `remote_address` ti the same value like `original_url`. Before I did this, I compared all te mirrors where `remote_address` was set.

**Comment by lunny** [comment]
> > Let's leave the issue open since it's possible a bug. And how did you do the workaround?
> 
> I set `remote_address` to the same value like `original_url`. Before I did this, I compared all te mirrors where `remote_address` was set.

After you fix it manually, does any mirrors remote_address become empty?

**Comment by wxiaoguang** [comment]
There are not only one issue report for the RemoteAddress problem (eg:  #29132)

I think " Add RemoteAddress to mirrors #26952 " is not robust enough.

**Comment by lunny** [comment]
> There are not only one issue report for the RemoteAddress problem (eg: #29132)
> 
> I think " Add RemoteAddress to mirrors #26952 " is not robust enough.

Yes, but maybe we can close this one or keep this open for some future refactor PRs about that?

**Comment by wxiaoguang** [comment]
> Yes, but maybe we can close this one or keep this open for some future refactor PRs about that?

But I just hope there could be a PR fixing the robust problem. It is critical and stops users from upgrading.

**Comment by richex-cn** [comment]
Same issue when I migrate to new version. I have manually fixed it using the following SQL:

```sql
UPDATE mirror m SET remote_address = (SELECT r.original_url FROM repository r WHERE r.id = m.repo_id);
```

**Comment by somera** [comment]
I didn't see this issue again after my manual fix in 2023.

**Comment by wxiaoguang** [comment]
> There are not only one issue report for the RemoteAddress problem (eg: [#29132](https://github.com/go-gitea/gitea/issues/29132))
> 
> I think " Add RemoteAddress to mirrors [#26952](https://github.com/go-gitea/gitea/pull/26952) " is not robust enough.

Yup, it is a bug of #26952 (the migration), so after the version upgrade, it's unlikely to happen again.

**Comment by wxiaoguang** [comment]
> There are not only one issue report for the RemoteAddress problem (eg: [#29132](https://github.com/go-gitea/gitea/issues/29132))
> 
> I think " Add RemoteAddress to mirrors [#26952](https://github.com/go-gitea/gitea/pull/26952) " is not robust enough.

@KN4CK3R maybe also related to Missing Source Repository URL in Migration Status (API vs UI) #36952

**Comment by somera** [comment]
Can be closed
